Enhancement-19322 add a session variable enable_pp_fuzzy

felixwluo opened the pull request:
# Proposed changes

Issue Number: #19322

## Problem summary

The new session variable enable_pp_fuzzy is in the fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java file. When it is true, the fragment will not be sent

## Checklist(Required)

* [x] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [x] Is this PR support rollback (If NO, please explain WHY)

## Further comments

Add a new session variable, not a big change.

15767714253 committed:
[UDF](demo) add new demo code for java udf (#19276)

Gabriel39 committed:
Revert "[refactor](function) ignore DST for function `from_unixtime` (#19151)" (#19333)

This reverts commit 9dd6c8f87b73db238bfd38fb1d76f3796910f398.

zy-kkk committed:
[fix](jdbc catalog) fix trino jdbc catalog varchar type err (#19298)

Commit:
Enhancement-19322 add a session variable enable_pp_fuzzy

Commit:
Increase the processing part of isPointQuery

felixwluo committed:
Merge branch 'apache:master' into dev-doris-enhancement-19322

1330571 committed:
[Enhancement](Broker Load) New progress manager for showing loading progress status (#19170)

This work is in the early stage, current progress is not accurate because the scan range will be too large
for gathering information, what's more, only file scan node and import job support new progress manager

## How it works

for example, when we use the following load query:
```
LOAD LABEL test_broker_load
(
	DATA INFILE("XXX")
	INTO TABLE `XXX`
        ......
)
```

Initial Progress: the query will call `BrokerLoadJob` to create job, then `coordinator` is called to calculate scan range and its location. 
Update Progress: BE will report runtime_state to FE and FE update progress status according to jobID and fragmentID

we can use `show load` to see the progress

PENDING:
```
         State: PENDING
      Progress: 0.00%
```

LOADING:
```
         State: LOADING
      Progress: 14.29% (1/7)
```

FINISH:
```
         State: FINISHED
      Progress: 100.00% (7/7)
```

At current time, full output of `show load\G` looks like:

```
*************************** 1. row ***************************
         JobId: 25052
         Label: test_broker
         State: LOADING
      Progress: 0.00% (0/7)
          Type: BROKER
       EtlInfo: NULL
      TaskInfo: cluster:N/A; timeout(s):250000; max_filter_ratio:0.0
      ErrorMsg: NULL
    CreateTime: 2023-05-03 20:53:13
  EtlStartTime: 2023-05-03 20:53:15
 EtlFinishTime: 2023-05-03 20:53:15
 LoadStartTime: 2023-05-03 20:53:15
LoadFinishTime: NULL
           URL: NULL
    JobDetails: {"Unfinished backends":{"5a9a3ecd203049bc-85e39a765c043228":[10080]},"ScannedRows":39611808,"TaskNumber":1,"LoadBytes":7398908902,"All backends":{"5a9a3ecd203049bc-85e39a765c043228":[10080]},"FileNumber":1,"FileSize":7895697364}
 TransactionId: 14015
  ErrorTablets: {}
          User: root
       Comment: 
```

## TODO:

1. The current partition granularity of scan range is too large, resulting in an uneven loading process for progress."
2. Only broker load supports the new Progress Manager, support progress for other query

Hastyshell committed:
[enhancement](load) merge single-replica related services as non-standalone  (#18421)

zy-kkk committed:
[typo](docs) add mac local dev docs (#19342)

* [typo](docs) add mac local dev docs

hello-stephen committed:
[tools](tpch) run mode like clickbench (#19339)

airborne12 committed:
[chore](build) Ignore clucene checks (#19353)

dataroaring reviewed (changes requested):
Please add a regression test.

kaka11chen committed:
[Fix](parquet-reader) Fix dict cols not be converted back to string type in some cases. (#19348)

Fix dict cols not be converted back to string type in some cases, which includes introduced by #19039.
For dict cols, we will convert dict cols to int32 type firstly, then convert back to string type after read block. 
The block will be reuse it, so it is necessary to convert it back.

felixwluo commented:
OK

felixwluo commented:
I think it's complicated. I'll make some changes

奕冷 ***@***.***>于2023年5月7日 周日11:16写道：

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java
> <https://github.com/apache/doris/pull/19350#discussion_r1186699508>:
>
> >          if (!isPointQuery) {
> -            if (enablePipelineEngine) {
> +            if (enablePipelineEngine && !enablePpFuzzy) {
>                  sendPipelineCtx();
> -            } else {
> +            } else if (!enablePpFuzzy) {
>                  sendFragment();
>              }
> -        } else {
> +        } else if (!enablePpFuzzy) {
>              OlapScanNode planRoot = (OlapScanNode) fragments.get(0).getPlanRoot();
>              Preconditions.checkState(planRoot.getScanTabletIds().size() == 1);
>              pointExec.setCandidateBackends(planRoot.getScanBackendIds());
>
> How about using
>
> if (enablePpFuzzy) {
>     return;
> }
>
> to skip the logic below?
> ------------------------------
>
> In fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java
> <https://github.com/apache/doris/pull/19350#discussion_r1186700724>:
>
> > @@ -352,6 +353,10 @@ public class SessionVariable implements Serializable, Writable {
>      @VariableMgr.VarAttr(name = ENABLE_PROFILE, needForward = true)
>      public boolean enableProfile = false;
>
> +    // if true, coordinator do not send planfragment to be, and just ends the query.
> +    @VariableMgr.VarAttr(name = ENABLE_PP_FUZZY)
>
> Does it need to forward to master?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/apache/doris/pull/19350#pullrequestreview-1415778393>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A7KEOOIVZ5CTTD6K2TKNGJ3XE4HX5ANCNFSM6AAAAAAXYBB7SU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

Yukang-Lian reviewed (commented):
Please rename this PR to \[enhancement\](fuzzy)add a session variable enable_pp_fuzzy

Commit:
[Enhancement-19322]variable ENABLE_PP_FUZZY forward,Simplify cancellation query logic

Commit:
Add comments to the Coordinator class
